exchanges/deribit/deribit_wrapper.go

```diff
@@ -0,0 +1,1303 @@
+			if err != nil {
+				return nil, err
+			}
+			resp = resp.Add(cp)
+		}
+	}
+	return resp, nil
+}
+
+// UpdateTradablePairs updates the exchanges available pairs and stores
+// them in the exchanges config
+func (d *Deribit) UpdateTradablePairs(ctx context.Context, forceUpdate bool) error {
+	assets := d.GetAssetTypes(false)
+	for x := range assets {
+		pairs, err := d.FetchTradablePairs(ctx, assets[x])
+		if err != nil {
+			return err
+		}
+		err = d.UpdatePairs(pairs, assets[x], false, forceUpdate)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// UpdateTickers updates the ticker for all currency pairs of a given asset type
+func (d *Deribit) UpdateTickers(_ context.Context, _ asset.Item) error {
+	return common.ErrFunctionNotSupported
+}
+
+// UpdateTicker updates and returns the ticker for a currency pair
+func (d *Deribit) UpdateTicker(ctx context.Context, p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	if !d.SupportsAsset(assetType) {
+		return nil, fmt.Errorf("%s: %w - %s", d.Name, asset.ErrNotSupported, assetType)
+	}
+	p, err := d.FormatExchangeCurrency(p, assetType)
+	if err != nil {
+		return nil, err
+	}
+	var instrumentID string
+	if assetType == asset.Futures {
+		instrumentID = d.formatFuturesTradablePair(p)
+	} else {
+		instrumentID = p.String()
+	}
+	var tickerData *TickerData
+	if d.Websocket.IsConnected() {
+		tickerData, err = d.WSRetrievePublicTicker(instrumentID)
+	} else {
+		tickerData, err = d.GetPublicTicker(ctx, instrumentID)
+	}
+	if err != nil {
+		return nil, err
+	}
+	resp := ticker.Price{
+		ExchangeName: d.Name,
+		Pair:         p,
+		AssetType:    assetType,
+		Ask:          tickerData.BestAskPrice,
+		AskSize:      tickerData.BestAskAmount,
+		Bid:          tickerData.BestBidPrice,
+		BidSize:      tickerData.BestBidAmount,
+		High:         tickerData.Stats.High,
+		Low:          tickerData.Stats.Low,
+		Last:         tickerData.LastPrice,
+		Volume:       tickerData.Stats.Volume,
+	}
+	err = ticker.ProcessTicker(&resp)
+	if err != nil {
+		return nil, err
+	}
+	return ticker.GetTicker(d.Name, p, assetType)
+}
+
+// FetchTicker returns the ticker for a currency pair
+func (d *Deribit) FetchTicker(ctx context.Context, p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	tickerNew, err := ticker.GetTicker(d.Name, p, assetType)
+	if err != nil {
+		return d.UpdateTicker(ctx, p, assetType)
+	}
+	return tickerNew, nil
+}
+
+// FetchOrderbook returns orderbook base on the currency pair
+func (d *Deribit) FetchOrderbook(ctx context.Context, currencyPair currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
+	ob, err := orderbook.Get(d.Name, currencyPair, assetType)
+	if err != nil {
+		return d.UpdateOrderbook(ctx, currencyPair, assetType)
+	}
+	return ob, nil
+}
+
+// UpdateOrderbook updates and returns the orderbook for a currency pair
+func (d *Deribit) UpdateOrderbook(ctx context.Context, p currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
```

Comment:
[ERROR] | ORDERBOOK | 01/02/2024 11:06:52 | Failed to get REST orderbook for Deribit BTC-FS-29MAR24_PERP future_combo. Error: cannot load book for exchange Deribit pair BTC-FS-29MAR24_PERP asset future_combo for Bids: price cannot be zero

Can you please check this?